assets/javascripts/discourse/components/rag-options.gjs

```diff
@@ -0,0 +1,70 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { Input } from "@ember/component";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import DTooltip from "discourse/components/d-tooltip";
+import I18n from "discourse-i18n";
+
+export default class RagOptions extends Component {
+  @tracked showIndexingOptions = false;
+
+  @action
+  toggleIndexingOptions(event) {
+    this.showIndexingOptions = !this.showIndexingOptions;
+    event.preventDefault();
+    event.stopPropagation();
+  }
+
+  get indexingOptionsText() {
+    return this.showIndexingOptions
+      ? I18n.t("discourse_ai.rag.options.hide_indexing_options")
+      : I18n.t("discourse_ai.rag.options.show_indexing_options");
+  }
+
+  <template>
+    {{#if @model.rag_uploads}}
+      <a
+        href="#"
+        class="rag-options__indexing-options"
+        {{on "click" this.toggleIndexingOptions}}
+      >{{this.indexingOptionsText}}</a>
```

Comment:
Can we use a button here? It's a lot of code just because we try to use an anchor for something which is not an anchor

Reply:
probably, will leave this to @awesomerobot to address in a followup, he probably wants to rethink some of the design here.